flytekit/core/python_function_task.py

```diff
@@ -145,11 +146,17 @@ def __init__(
                 )
         self._task_function = task_function
         self._execution_mode = execution_mode
+        self._output_entity_hint = output_entity_hint
+        assert (self._output_entity_hint is None) or (self._execution_mode == self.ExecutionBehavior.DYNAMIC), "output_entity_hint should only be specified on dynamic tasks."
         self._wf = None  # For dynamic tasks
 
     @property
     def execution_mode(self) -> ExecutionBehavior:
         return self._execution_mode
+    
+    @property
+    def output_entity_hint(self) -> ExecutionBehavior:
```

Comment:
Why is this of type `ExecutionBehavior` ? Should it be `FlyteLocalEntity` ? Same below when you pass it in `task.py`